docs/appendices/release-notes/unreleased.rst

```diff
@@ -268,5 +268,8 @@ Performance improvements
 Fixes
 =====
 
+- Fixed an issue where :ref:`drop snapshots <ref-drop-snapshot>` would not
+  delete all the related data.
```

Comment:
Maybe clarify that this affected Azure repositories? Or was it also the case for S3?